core/src/main/java/org/apache/druid/data/input/AbstractInputSource.java

```diff
@@ -53,12 +54,13 @@ protected InputSourceReader formattableReader(
       InputRowSchema inputRowSchema,
       InputFormat inputFormat,
       @Nullable File temporaryDirectory
-  )
+  ) throws IOException
   {
     throw new UnsupportedOperationException("Implement this method properly if needsFormat() = true");
   }
 
   protected InputSourceReader fixedFormatReader(InputRowSchema inputRowSchema, @Nullable File temporaryDirectory)
+      throws IOException
```

Comment:
TeamCity is flagging this: `The declared exception IOException is never thrown in this method, nor in its derivables`

Reply:
Ah yeah, nothing implements this method yet, I was preemptively adding it because `InputSource.createSplits` is allows to throw an `IOException` and readers will most likely be calling this method. I can remove for now, though I suspect it will likely need to be added back later.